Coffee-Code-Philly-Accelerator/coffee_and_code_philly_dot_com
# Feat: Quiz Carousel

### Description
This PR implements a quiz carousel functionality, allowing users to navigate through quiz questions without triggering form submission until the final submission button is clicked. The final question presents a code of conduct agreement. This agreement or any other question in the quiz is not currently validated. This PR is closely related to #15, which should be completed subsequent to this one's closing.

Closes #21 

### Changes Made
- Updated `OnboardingQuiz` component to properly manage the quiz state and handle next question navigation.
- Removed variant of `Choice` component for variable/custom text inputs. These custom inputs were adding unneccessary complexity to the implementation of the `OnboardinQuiz` component and its children.
- Implemented event handling to prevent form submission when navigating through questions.
- Implemented event handling for the completion of the quiz (with no validation).

### Features
- [x] Display multiple quiz questions.
- [x] Navigate through questions using "Next Question" button.
- [x] Form submission only triggered by the "Submit" button.

### Screenshots
_(Optional: Add screenshots or GIFs to showcase the changes)_
<img width="679" alt="Screen Shot 2024-07-12 at 2 30 56 PM" src="https://github.com/user-attachments/assets/395a135b-fa56-4f64-ae20-d53dc585a92c">
<img width="683" alt="Screen Shot 2024-07-12 at 2 31 10 PM" src="https://github.com/user-attachments/assets/bd5fb562-112a-42be-a9ab-b2f1095b6794">


### Testing
1. Navigate through quiz questions using the "Next Question" button.
2. Ensure that the form is not submitted until the "Submit" button is clicked.
3. Verify that the answers are collected and stored correctly.

### Checklist
- [x] Changes have been reviewed by at least one team member.

### Related Issues

#15 , #16 , #21 

### Additional Notes
This component needs to be styled. It has very bare-bones styles as is.

Thank you for reviewing this PR. Your feedback is appreciated!
